ParkingFinder/repositories/check_in_parking_space_repository.py

```diff
@@ -0,0 +1,126 @@
+from ParkingFinder.base.errors import NotFound
+from ParkingFinder.mappers.parking_space_mapper import CheckInParkingSpaceMapper
+from ParkingFinder.tables.check_in_parking_space import CheckInParkingSpace
+
+
+class CheckInParkingSpaceRepository(object):
+
+    @staticmethod
+    @coroutine
+    def read_one(user_id):
+        """
+
+        :param str user_id:
+        :return ParkingSpace parking_space:
+        """
+        with create_session() as session:
+            parking_space = session.query(CheckInParkingSpace).filter(
+                CheckInParkingSpace.user_id == user_id).one()
+            entity = CheckInParkingSpaceMapper.to_entity(record=parking_space)
+            raise Return(entity)
+
+    @classmethod
+    @coroutine
+    def upsert(cls, parking_space):
+        """
+
+        Update/Insert parking space, if user already have parking space, update user's information
+        otherwise, insert a new parking space, and parking_space have to be a ParkingSpace entity
+        with all required fields
+
+        :param ParkingSpace parking_space:
+        :return ParkingSpace:
+        :raises AssertionError: the parking_space provided doesn't have user_id
+        :raises ValidationError: the parking_space provided missing required fields
+        """
+        assert parking_space.user_id
+
+        try:
+            _parking_space = yield cls._update(
+                parking_space=parking_space,
+                latitude=parking_space.latitude,
+                longitude=parking_space.longitude,
+                created_at=parking_space.create_at,
+                level=parking_space.level,
+                description=parking_space.description
+            )
+            raise Return(_parking_space)
+        except NoResultFound:
+            _parking_space = yield cls._insert(parking_space)
+            raise Return(_parking_space)
+
+    @staticmethod
+    @coroutine
+    def remove(user_id):
+        """
+        Remove a parking_space using user_id
+
+        :param str user_id:
+        :return ParkingSpace: deleted parking space
+        """
+        with create_session() as session:
+            rows = session.query(CheckInParkingSpace).filter(
+                CheckInParkingSpace.user_id == user_id
+            ).delete()
+            if rows == 0:
+                raise NoResultFound
+
+            raise Return(rows)
+
+
+    @classmethod
+    @coroutine
+    def _update(cls,
+                parking_space,
+                latitude,
+                longitude,
+                created_at,
+                level,
+                description):
+        """
+
+        :param ParkingSpace parking_space:
+        :return ParkingSpace:
+        """
+        with create_session() as session:
+            _parking_space = session.query(CheckInParkingSpace).filter(
+                CheckInParkingSpace.user_id == parking_space.user_id
+            ).one()
+
+            if latitude:
```

Comment:
There is no a case without latitude and longitude. .
fix this or _update with default value.
or just pass parkingspace.
there is not needed to pass both entity and argument.